File: coding/24th February 2025/Longest Palindrome.java
//Longest Palindrome
class Solution 
{
    public int longestPalindrome(String s) 
    {
        if(s.length() == 1)   return 1;
        int odd = 0;
        HashMap<Character, Integer> map = new HashMap<Character, Integer>();
        for(char ch : s.toCharArray())
        {
            if(map.containsKey(ch))
                map.put(ch, map.get(ch) + 1);
            else
                map.put(ch, 1);
            if(map.get(ch) % 2 != 0)
                odd++;
            else
                odd--;
        }
        if(odd > 1)   return s.length() - odd + 1;
        return s.length();
    }
}

File: Coding in Python/03rd April 2026/Longest Palindrome.py
#Longest Palindrome
class Solution:
    def longestPalindrome(self, s: str) -> int:
        odd = 0
        map = {}
        for ch in s:
            if ch in map:
                map[ch] += 1
            else:
                map[ch] = 1
            if map[ch] % 2 == 0:
                odd -= 1
            else:
                odd += 1
        if odd > 1:
            return len(s) - odd + 1
        return len(s)
